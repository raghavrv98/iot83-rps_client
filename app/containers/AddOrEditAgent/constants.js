/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
