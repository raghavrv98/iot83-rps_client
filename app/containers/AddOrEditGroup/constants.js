/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
