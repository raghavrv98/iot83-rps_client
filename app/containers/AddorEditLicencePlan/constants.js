/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";