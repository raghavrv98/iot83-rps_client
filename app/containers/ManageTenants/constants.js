/*
 *
 * ManageTenants constants
 *
 */

export const DEFAULT_ACTION = "app/ManageTenants/DEFAULT_ACTION";

export const GET_TENANTS = "app/ManageTenants/GET_TENANTS";
export const GET_TENANTS_SUCCESS = "app/ManageTenants/GET_TENANTS_SUCCESS";
export const GET_TENANTS_FAILURE = "app/ManageTenants/GET_TENANTS_FAILURE";


export const DELETE_TENANT = "app/ManageTenants/DELETE_TENANT";
export const DELETE_TENANT_SUCCESS = "app/ManageTenants/DELETE_TENANT_SUCCESS";
export const DELETE_TENANT_FAILURE = "app/ManageTenants/DELETE_TENANT_FAILURE";
