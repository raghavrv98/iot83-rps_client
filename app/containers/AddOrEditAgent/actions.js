/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
