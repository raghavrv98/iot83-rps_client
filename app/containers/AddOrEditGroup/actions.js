/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
