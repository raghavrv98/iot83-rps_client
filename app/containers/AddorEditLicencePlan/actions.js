/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}