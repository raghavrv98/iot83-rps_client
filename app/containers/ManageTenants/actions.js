/*
 *
 * ManageTenants actions
 *
 */

import { DEFAULT_ACTION,GET_TENANTS,DELETE_TENANT } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function getTenants(){
  return{
    type:GET_TENANTS
  }
}

export function deleteHandler(id){
  return {
    type:DELETE_TENANT,
    id
  }
}