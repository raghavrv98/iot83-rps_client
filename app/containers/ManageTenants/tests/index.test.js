import React, { Component } from 'react';
import { shallow } from 'enzyme';
import history from "utils/history";
import { getTenants, deleteHandler } from "../actions";
import { GET_TENANTS, DELETE_TENANT } from "../constants";
import { ManageTenants, mapDispatchToProps } from "../index";

const tenantsList = [
    {
      _id: "9bed0d5ca79011e9863a0242ac110016",
      createdAt: 1563257514466,
      name: "edfedeefdf",
      other: "fvcerdfedfv",
      overallGeometry: "dfe3dfedc",
      plantId: "f29b79c09f0911e986810242ac110011",
      processFluid: "efe3dcfed",
      updatedAt: 1563257547048
    }
  ];

describe(' < ManageTenants />', () => {
    const match = {
        params: {
            id: ""
        }
    };
    it('Expect to have unit tests specified', () => {
        expect(true).toEqual(true);
    });

    it('match snapshots', () => {
        const wrap = shallow(<ManageTenants match={match} getTenants={getTenants} history={history}/>)
        expect(wrap).toMatchSnapshot()
    })

    it("test componentDidMount", async () => {
        const match = {
            params: {
                id: "demoId"
            }
        };
        const getTenants = jest.fn();
        spyOn(ManageTenants.prototype, "componentDidMount").and.callThrough();
        shallow(
            <ManageTenants
                history={history}
                match={match}
                getTenants={getTenants}
            />
        );
        expect(ManageTenants.prototype.componentDidMount).toHaveBeenCalledTimes(1);
        expect(getTenants).toHaveBeenCalledTimes(0);
    });

    it('should test component when recieve props', () => {
        spyOn(ManageTenants.prototype, 'componentWillReceiveProps').and.callThrough();
        const wrapper = shallow(<ManageTenants getTenants={getTenants} history={history} list={undefined} listError={undefined} deleteSuccess={undefined} deleteFailure={undefined} />);
        expect(wrapper).toBeDefined();
        wrapper.setState({ userList: tenantsList });
        wrapper.setProps({ list: tenantsList, listError: "No Error Found.", deleteSuccess: "bb157d6738bb4886b77fd25635301010", deleteFailure: "No Error Found in delete." });
        expect(ManageTenants.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
      })

    // it("clicking add button trigger onClick event", async () => {
    //     let expectedCount = history.length + 2;
    //     const wrapper = shallow(<ManageTenants getTenants={getTenants} history={history} />);
    //     let addButton = wrapper.find("button.btn.btn-create");
    //     expect(addButton.length).toBe(1);
    //     addButton.simulate('click');
    //     expect(history.length).toEqual(expectedCount);
    //     expect(history.location.pathname).toEqual("/addOrEditTenant");
    // });

    it("Function testing on mapDispatchToProps", () => {
        const mockParams = {
          params: {
            id: "demo"
          }
        };
        const dispatch = jest.fn();
        mapDispatchToProps(dispatch).getTenants();
        expect(dispatch.mock.calls[0][0]).toEqual({
          type: GET_TENANTS,
        });
        mapDispatchToProps(dispatch).deleteHandler(
          mockParams.params.id
        );
        expect(dispatch.mock.calls[1][0]).toEqual({
          type: DELETE_TENANT,
          id: "demo"
        });
    });
});