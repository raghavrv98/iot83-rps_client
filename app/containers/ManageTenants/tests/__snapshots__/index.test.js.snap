// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` < ManageTenants /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      ManageTenants
    </title>
    <meta
      content="Description of ManageTenants"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p>
        <FormattedMessage
          defaultMessage="Manage Tenants"
          id="app.common.messages.manageTenants"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="All Available Tenants"
          id="app.containers.ManageTenants.title"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <SkeletonLoader
    mode="fullView"
    skeleton="skeletonReactTable"
  />
</div>
`;
